tests/reference/asr-test_list3-5f4d2a8.stderr

```diff
@@ -1,5 +1,5 @@
-semantic error: Type mismatch in 'pop', List index should be of integer type
- --> tests/errors/test_list3.py:5:15
+semantic error: For now pop() takes no arguments
```

Comment:
These errors weren't there before. I would like to go back to the original errors. ``pop`` should take arguments and work with them.

Reply:
Added a commit to allow `pop` with position argument. Does not seem to work correctly with nested iterables.
```python
from lpython import i32, f64

def test_list_pop():
    l1: list[tuple[i32, f64]]
    x: tuple[i32, f64]

    l1 = [(1, 1.0), (2, 4.0)]
    x = l1.pop(0)
    print(x)        # incorrect
    print(l1)       # correct

test_list_pop()
```
```
(2, 4.00000000000000000e+00)
[(2, 4.00000000000000000e+00)]
```

Also, errors correctly caught in `verify_args` lead to an `LCompilersException`. It does not seem related to this specific PR, though.